src/main/flight/autopilot.c

```diff
@@ -91,14 +196,172 @@ void altitudeControl(float targetAltitudeCm, float taskIntervalS, float vertical
+        
+                // scale and filter - filter cutoffs vary during the startup phase
+                float pidD = velocity * positionPidCoeffs.Kd;
+                pt1FilterUpdateCutoff(&latLong->velocityLpf, posHold.pt1Gain);
+                pidD = pt1FilterApply(&latLong->velocityLpf, pidD);
+    
+                float pidA = acceleration * positionPidCoeffs.Kd;
+                pt2FilterUpdateCutoff(&latLong->accelerationLpf, posHold.pt1Gain);
+                pidA = pt2FilterApply(&latLong->accelerationLpf, pidA);
+        
+                // limit sum of D and A because otherwise can be too aggressive when starting at speed
+                const float maxDAAngle = 35.0f; // limit in degrees; arbitrary.  20 is a bit too low, allows a lot of overshoot
+                // an angle of more than 35 degrees is achieved as P and I grow
+                // ** todo = should this be half of the user-configurable angle_limit?  Or fixed?
+                const float pidDA = constrainf(pidD + pidA, -maxDAAngle, maxDAAngle);
+    
+                // ** PID Sum **
+                float pidSum = pidP + pidI + pidDA;
+    
+                // terminate initial startup behaviour separately for latitude and longitude controllers
+                // the position target is reset when pidSum crosses zero
+                // this enhances the smoothness of the transition from stick input back to position hold - there is no sharp change in pidSum
+                if (latLong->isStarting && latLong->pidSum * pidSum < 0.0f) { // pidsum ns has reversed sign
+                    resetPositionControlParams(latLong);
+                    if (i == 0) {
+                        currentTargetLocation.lat = gpsSol.llh.lat;  // can we simplify this within the loop?
+                    } else {
+                        currentTargetLocation.lon = gpsSol.llh.lon;
+                    }
+                    latLong->distance = 0.0f;
+                    latLong->isStarting = false;
+                }
+                latLong->pidSum = pidSum;
+    
+                // Debugs... distances in cm, angles in degrees * 10, velocities cm/2
+                if (gyroConfig()->gyro_filter_debug_axis == i) {
+                    DEBUG_SET(DEBUG_AUTOPILOT_POSITION, 0, lrintf(distanceCm));
+                    DEBUG_SET(DEBUG_AUTOPILOT_POSITION, 1, lrintf(latLong->distance));
+                    DEBUG_SET(DEBUG_AUTOPILOT_POSITION, 2, lrintf(latLong->pidSum * 10));
+                    DEBUG_SET(DEBUG_AUTOPILOT_POSITION, 4, lrintf(pidP * 10));
+                    DEBUG_SET(DEBUG_AUTOPILOT_POSITION, 5, lrintf(pidI * 10));
+                    DEBUG_SET(DEBUG_AUTOPILOT_POSITION, 6, lrintf(pidDA * 10));
+                }
+            }
+        }
+        // ** Rotate pid Sum to quad frame of reference, into pitch and roll **
+        float headingRads = DECIDEGREES_TO_RADIANS(attitude.values.yaw); // will be constrained to +/-pi in sin_approx()
```

Comment:
(sin is defined for real argument, You are stating something obvious)
```suggestion
        float headingRads = DECIDEGREES_TO_RADIANS(attitude.values.yaw);
```

Reply:
Yes it was a reminder to myself, since previously I did wrapping to +/- 180 here, then realised it wasn't needed.
Comment removed.